### PR TITLE
Add some more throttler tests

### PR DESCRIPTION
**Pull request by arinwt**
Based on https://github.com/microsoft/FluidFramework/pull/6909#discussion_r681265873, added some more unit tests to throttler and pulled out the functions to be generalized for both exponential and linear.

### CODE REVIEW

File: packages/runtime/container-runtime/src/test/throttler.spec.ts [hunk 1]
```diff
@@ -134,4 +126,202 @@ describe("Throttler", () => {
...
+                clock.tick(i % 3 === 2 ? remainingTicks : oneThirdTicks);
+            }
+            assert.strictEqual(getDelayAndTick(), 40);
+
+            // This time fail instantly, giving a later delay.
+            assert.strictEqual(getDelayAndTick(), 80);
+        });
+
+        it("Should stop increasing number of attempts after max", () => {
+            for (let i = 0; i < 12; i++) {
```
Comment (andre4i) [inline packages/runtime/container-runtime/src/test/throttler.spec.ts:217]: perhaps `const` the value 12 with a comment about how it came to be (I assume it's num attempts within `maxDelayMs` timeframe). Same comment for line 308.

File: packages/runtime/container-runtime/src/test/throttler.spec.ts [hunk 2]
```diff
@@ -134,4 +126,202 @@ describe("Throttler", () => {
...
+                getDelayAndTick();
+                assert.strictEqual(throttler.numAttempts, i + 1, `loop 1; iteration ${i}`);
+            }
+            for (let i = 0; i < 100; i++) {
```
Comment (andre4i) [inline packages/runtime/container-runtime/src/test/throttler.spec.ts:221]: Most likely I'm advocating for some premature optimization wrt to how long it takes to run the tests :)

But do we really need to go to 100 in this loop? This is traversing an array (small one though) 100 times, right?

**Review by andre4i — APPROVED** [review]
👍

File: packages/runtime/container-runtime/src/test/throttler.spec.ts (hunk 2 above)
Comment (arinwt) [reply packages/runtime/container-runtime/src/test/throttler.spec.ts:221]: It probably doesn't need to go to 100.. I was thinking about potential edge cases: off by 1, where it slides one attempt out of the window after many attempts. I didn't actually calculate it out though, which means that 100 is very arbitrary, and maybe not even sufficient.

File: packages/runtime/container-runtime/src/test/throttler.spec.ts (hunk 1 above)
Comment (arinwt) [reply packages/runtime/container-runtime/src/test/throttler.spec.ts:217]: I decided to go ahead and consolidate all the tests, let me know what you think. Now it more clearly shows the calculation for that number:
```ts
const expectedMaxAttempts = expectedDelays.length;
```

Comment (andre4i) [reply packages/runtime/container-runtime/src/test/throttler.spec.ts]: Looks great

**Review by andre4i — APPROVED** [review]

**Pull request merged**